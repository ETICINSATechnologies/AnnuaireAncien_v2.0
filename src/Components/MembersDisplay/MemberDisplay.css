.MemberDisplay .positionYear {
    /*width: 10vw;*/
}
