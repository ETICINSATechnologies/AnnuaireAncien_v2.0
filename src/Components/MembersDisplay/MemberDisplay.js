import React, {Component} from 'react';
import './MemberDisplay.css';

class MemberDisplay extends Component {
    constructor(props) {
        super(props);
        let propertiesName = {
            id: null,
            lastName: 'nom',
            firstName: 'prénom',
            positionLabel: 'poste',
            positionYear: 'année'
        };
        this.state = {
            header: this.props.header || false,
            propertiesName: propertiesName,
            properties: this.getMemberProperties(propertiesName)
        };
    }

    getPositionLabel() {
        try {
            if (this.props.info.positions[0].label)
                return this.props.info.positions[0].label;
        } catch (e) {
            if (e instanceof TypeError) return null;
        }
    }

    getPositionYear() {
        try {
            if (this.props.info.positions) {
                if (this.props.info.positions[0].year)
                    return this.props.info.positions[0].year;
                return 2018;
            }
        } catch (e) {
            if (e instanceof TypeError) return 2018;
        }
    }

    getMemberProperties(propertiesName) {
        let newProperties = {};
        Object.keys(propertiesName).forEach((propertyName) => {
            if (this.props.info.hasOwnProperty(propertyName)) {
                newProperties[propertyName] = this.props.info[propertyName];
            }
        });
        newProperties.positionLabel = this.getPositionLabel();
        newProperties.positionYear = this.getPositionYear();

        return newProperties;
    }

    render() {
        return (
            <tr className={`MemberDisplay ${this.state.properties.id}`} onClick={() => this.props.onClick(this.state.properties.id)}>
                {this.state.header
                    ?
                    <React.Fragment>
                        <th className='firstName'>{this.state.propertiesName.firstName}</th>
                        <th className='lastName'>{this.state.propertiesName.lastName}</th>
                        <th className='positionLabel'>{this.state.propertiesName.positionLabel}</th>
                        <th className='positionYear'>{this.state.propertiesName.positionYear}</th>
                    </React.Fragment>
                    :
                    <React.Fragment>
                        <td className='firstName'>{this.state.properties.firstName}</td>
                        <td className='lastName'>{this.state.properties.lastName}</td>
                        <td className='positionLabel'>{this.state.properties.positionLabel}</td>
                        <td className='positionYear'>{this.state.properties.positionYear}</td>
                    </React.Fragment>
                }
            </tr>
        );
    }
}

export default MemberDisplay;
