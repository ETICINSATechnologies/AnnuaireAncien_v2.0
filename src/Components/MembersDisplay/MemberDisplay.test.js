import React from 'react';
import MemberDisplay from './MemberDisplay';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';


Enzyme.configure({adapter: new Adapter()});

describe( 'MemberDisplay component', () => {
    it('renders without crashing', () => {
            const tree = renderer.create(<MemberDisplay />).toJSON();
            expect(tree).toMatchSnapshot();
        }
    );

    it('MemberDisplay unit test', () => {
        expect(Enzyme.shallow(<MemberDisplay/>)
            .find('.MemberDisplay')
            .length
        ).toBe(1)
    });

});