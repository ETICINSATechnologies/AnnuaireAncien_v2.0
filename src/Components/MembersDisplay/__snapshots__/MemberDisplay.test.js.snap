// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberDisplay component renders without crashing 1`] = `
<tr
  className="MemberDisplay undefined"
  onClick={[Function]}
>
  <td
    className="firstName"
  />
  <td
    className="lastName"
  />
  <td
    className="positionLabel"
  />
  <td
    className="positionYear"
  >
    2018
  </td>
</tr>
`;
